osu.Game/Online/API/Requests/GetUserBeatmapsRequest.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Collections.Generic;
+
+namespace osu.Game.Online.API.Requests
+{
+    public class GetUserBeatmapsRequest : APIRequest<List<GetBeatmapSetsResponse>>
+    {
+        private readonly long userId;
+        private readonly int offset;
+        private readonly string type;
+
+        public GetUserBeatmapsRequest(long userId, BeatmapSetType type, int offset = 0)
+        {
+            this.userId = userId;
+            this.offset = offset;
+
+            switch (type)
+            {
+                case BeatmapSetType.Favourite:

```

Comment:
can we not just do this correctly as i described in discord?

add nuget package [Humanizer](https://www.nuget.org/packages/Humanizer) and use extension method `.Underscore()` directly in the `Target` string property